tests/by-util/test_rm.rs

```diff
@@ -1078,3 +1078,80 @@ fn test_rm_recursive_long_path_safe_traversal() {
     // Verify the directory is completely removed
     assert!(!at.dir_exists("rm_deep"));
 }
+
+#[cfg(all(not(windows), feature = "chmod"))]
+#[test]
+fn test_rm_directory_not_executable() {
+    // Test from GNU rm/rm2.sh
+    // Exercise code paths when directories have no execute permission
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    // Create directory structure: a/0, a/1/2, a/2, a/3, b/3
+    at.mkdir_all("a/0");
+    at.mkdir_all("a/1/2");
+    at.mkdir("a/2");
+    at.mkdir("a/3");
+    at.mkdir_all("b/3");
+
+    // Remove execute permission from a/1 and b
+    scene.ccmd("chmod").arg("u-x").arg("a/1").succeeds();
+    scene.ccmd("chmod").arg("u-x").arg("b").succeeds();
+
+    // Try to remove both directories recursively - this should fail
+    let result = scene.ucmd().args(&["-rf", "a", "b"]).fails();
+
+    // Check for expected error messages
+    let stderr = result.stderr_str();
+    assert!(
+        (stderr.contains("rm: cannot remove 'a/1': Directory not empty")
+            && stderr.contains("rm: cannot remove 'a': Directory not empty"))
+            || (stderr.contains("rm: cannot remove 'a/1/2': Permission denied")
+                && stderr.contains("rm: cannot remove 'b/3': Permission denied"))
```

Comment:
What's the reason for the OR? I would expect that the command always returns the same error messages.